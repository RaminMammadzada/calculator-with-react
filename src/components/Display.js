import React from 'react';
import PropTypes from 'prop-types';

const Display = props => {
  let { result } = props;

  if (result === null) {
    result = 0;
  }

  return (
    <div className="display">
      <p>{result}</p>
    </div>
  );
};

Display.defaultProps = {
  result: '0',
};

Display.propTypes = {
  result: PropTypes.string,
};

export default Display;
