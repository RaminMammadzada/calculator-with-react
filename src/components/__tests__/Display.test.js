import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Display from '../Display';

describe('Display', () => {
  it('should render the Display component', () => {
    const tree = ReactTestRenderer.create(<Display />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should create a Display component', () => {
    render(<Display result="0" />);
    const element = screen.getByText('0');
    expect(element).toBeInTheDocument();
  });
});
