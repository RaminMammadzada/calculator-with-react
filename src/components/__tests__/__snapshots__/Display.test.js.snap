// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should render the Display component 1`] = `
<div
  className="display"
>
  <p>
    0
  </p>
</div>
`;
